python_modules/dagster/dagster/_core/definitions/auto_materialize_sensor_definition.py

```diff
@@ -1,12 +1,102 @@
 from typing import Any, Mapping, Optional, cast
 
+import pendulum
+
+import dagster._check as check
 from dagster._annotations import experimental
+from dagster._core.asset_graph_view.asset_graph_view import AssetGraphView, TemporalContext
 from dagster._core.definitions.asset_selection import CoercibleToAssetSelection
+from dagster._core.definitions.data_time import CachingDataTimeResolver
+from dagster._core.definitions.data_version import CachingStaleStatusResolver
+from dagster._core.definitions.declarative_scheduling.scheduling_condition_evaluator import (
+    SchedulingConditionEvaluator,
+)
+from dagster._core.definitions.run_request import SensorResult
+from dagster._utils.caching_instance_queryer import CachingInstanceQueryer
 
 from .asset_selection import AssetSelection
-from .sensor_definition import DefaultSensorStatus, SensorDefinition, SensorType
+from .sensor_definition import (
+    DefaultSensorStatus,
+    SensorDefinition,
+    SensorEvaluationContext,
+    SensorType,
+)
 from .utils import check_valid_name, normalize_tags
 
+EXECUTABLE_AMP_SENSOR_DESC = "Experimental user space scheduling condition evaluation sensor."
+
+
+def evaluate_scheduling_conditions(context: SensorEvaluationContext):
+    from dagster._core.definitions.asset_daemon_context import build_run_requests
+    from dagster._daemon.asset_daemon import (
+        asset_daemon_cursor_from_instigator_serialized_cursor,
+        asset_daemon_cursor_to_instigator_serialized_cursor,
+    )
+
+    asset_graph = check.not_none(context.repository_def).asset_graph
+
+    instance_queryer = CachingInstanceQueryer(
```

Comment:
I think 38-55 should be a helper function on `AssetGraphView` to construct it from an existing instance. We can do that in a follow up.